### DISCUSSION TITLE
useBalance() Function for NFTS token ERC721A?

### DISCUSSION THREAD

**Discussion by LeoAnt02** [wevm/wagmi#780] [General]
I'm trying to fetch the data of a wallet 
I want to know if a specific NFT ERC-721A is in my wallet 

    const { address, isDisconnected, isConnected } = useAccount()

    const { data, isError } = useBalance({
        addressOrName: address,
        chainId:4,
        token: '0xe2200bEeeb26A2FCd67973d5443201B7d6a296Ba',
      })
      
      
 When I remove the token, i can see my Rinkeby eth, but when I add it, I see nothing

**Comment by LeoAnt02** [comment]
I just tried with a test LINK token on the rinkeby testnet, and it's working with ERC-20 TOKEN 
0x01BE23585060835E02B77ef475b0Cc51aA1e0709
But definitly not with NFTs ERC721A token 

Is this me ? OR am I missing something ? 

Would love to have help on this

**Reply by tmm** [reply]
`useBalance` only works for ether balances ERC20 tokens since `decimals` isn't a part of the ERC721 interface.

**Reply by LeoAnt02** [reply]
Any idea on an alternative ? 
On how to read an NFT from a wallet

**Reply by tmm** [reply]
Should be able to do something like this with [`useContractReads`](https://wagmi.sh/docs/hooks/useContractReads):

```tsx
import { useContractReads, erc721ABI } from 'wagmi'

const contractConfig = {
  addressOrName: '0xe2200bEeeb26A2FCd67973d5443201B7d6a296Ba',
  chainId: 4,
  contractInterface: erc721ABI,
}

function App() {
  const { data, isError, isLoading } = useContractReads({
    contracts: [
      {
        ...contractConfig,
        functionName: 'balanceOf',
        args: '<address to check balance for>',
      },
      { ...contractConfig, functionName: 'symbol' },
    ],
  })
}
```

**Reply by LeoAnt02** [reply]
Thanks for the feedback 

What's the purpose of the line 

      { ...contractConfig, functionName: 'symbol' },
      
And how can I retrieve the value of the BalanceOf ?

**Reply by tmm** [reply]
It fetches the symbol for the ERC721 so you can display it with the balance (similar to how `useBalance` returns the symbol).

If you only want the balance, you can use `useContractRead` instead:

```ts
import { useContractRead, erc721ABI } from 'wagmi'

function App() {
  const { data, isError, isLoading } = useContractRead({
    addressOrName: '0xe2200bEeeb26A2FCd67973d5443201B7d6a296Ba',
    chainId: 4,
    contractInterface: erc721ABI,
    functionName: 'balanceOf',
    args: '<address to check balance for>',
  })
}
```

**Reply by LeoAnt02** [reply]
Yes, I do use the useContractRead() 

I added isSuccess 

    const { data, isError, isSuccess } = useContractRead({
        addressOrName: '0xe2200bEeeb26A2FCd67973d5443201B7d6a296Ba',
        chainId: 4,
        contractInterface: NFT_Student,
        functionName: 'balanceOf',
        args: address,
               
        onSuccess(data) {
            var open = true;
            console.log('Success', data)
        },
    })
    
I have isSuccess = true when I'm connecting to a wallet with the NFT 
and isSuccess = false when connecting to a wallet without the NFT

Is this the proper way to do it ?

Now everything seems to work, but I want to make sure it's the right way

**Reply by tmm** [reply]
`onSuccess` is fired when the query request is successful. You still need to inspect `data` to see what the balance is.

**Reply by LeoAnt02** [reply]
I have changed the metadata of my collection and now everything seems broken, I don't understand why I have this error


TypeError: abi.map is not a function
    at new Interface (interface.ts:100 :1)
    at getInterface (index.ts:820:1)
    at new BaseContract (index.ts:662:1)
    at new Contract (index.ts:1140:1)
    at getContract (wagmi-core.esm.js:290:1)
    at readContract (wagmi-core.esm.js:355:1)
    at Object.queryFn$7 [as queryFn] (wagmi.esm.js:1365:1)
    at Object.fetchFn [as fn] (query.mjs:249:1)
    at run (retryer.mjs:97:1)
    at createRetryer (retryer.mjs:143:1)
    
    
    
   I didnt change the code above, just added external library like react-dom-route etc... 
   
   Any idea on how to have a successful read from the contract ?
